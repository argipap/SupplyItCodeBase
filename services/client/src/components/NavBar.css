.link:hover {
  cursor: pointer;
  background-color: #292929;
  color: #f5f5f5;
}




/* 
.navbar {
  background-color: #ffffff;
}
.navbar .navbar-brand {
  color: #232424;
}
.navbar .navbar-brand:hover,
.navbar .navbar-brand:focus {
  color: #ffffff;
}
.navbar .navbar-text {
  color: #232424;
}
.navbar .navbar-text a {
  color: #ffffff;
}
.navbar .navbar-text a:hover,
.navbar .navbar-text a:focus {
  color: #ffffff; 
}
.navbar .navbar-nav .nav-link {
  color: #232424;
  border-radius: .25rem;
  margin: 0 0.25em;
}
.navbar .navbar-nav .nav-link:not(.disabled):hover,
.navbar .navbar-nav .nav-link:not(.disabled):focus {
  color: #ffffff;
}
.navbar .navbar-nav .dropdown-menu {
  background-color: #ffffff;
  border-color: #de6b48;
}
.navbar .navbar-nav .dropdown-menu .dropdown-item {
  color: #232424;
}
.navbar .navbar-nav .dropdown-menu .dropdown-item:hover,
.navbar .navbar-nav .dropdown-menu .dropdown-item:focus,
.navbar .navbar-nav .dropdown-menu .dropdown-item.active {
  color: #ffffff;
  background-color: #de6b48;
}
.navbar .navbar-nav .dropdown-menu .dropdown-divider {
  border-top-color: #de6b48;
}
.navbar .navbar-nav .nav-item.active .nav-link,
.navbar .navbar-nav .nav-item.active .nav-link:hover,
.navbar .navbar-nav .nav-item.active .nav-link:focus,
.navbar .navbar-nav .nav-item.show .nav-link,
.navbar .navbar-nav .nav-item.show .nav-link:hover,
.navbar .navbar-nav .nav-item.show .nav-link:focus {
  color: #ffffff;
  background-color: #de6b48;
}
.navbar .navbar-toggle {
  border-color: #de6b48;
}
.navbar .navbar-toggle:hover,
.navbar .navbar-toggle:focus {
  background-color: #de6b48;
}
.navbar .navbar-toggle .navbar-toggler-icon {
  color: #232424;
}
.navbar .navbar-collapse,
.navbar .navbar-form {
  border-color: #232424;
}
.navbar .navbar-link {
  color: #232424;
}
.navbar .navbar-link:hover {
  color: #ffffff;
}

@media (max-width: 575px) {
  .navbar-expand-sm .navbar-nav .show .dropdown-menu .dropdown-item {
    color: #232424;
  }
  .navbar-expand-sm .navbar-nav .show .dropdown-menu .dropdown-item:hover,
  .navbar-expand-sm .navbar-nav .show .dropdown-menu .dropdown-item:focus {
    color: #ffffff;
  }
  .navbar-expand-sm .navbar-nav .show .dropdown-menu .dropdown-item.active {
    color: #ffffff;
    background-color: #de6b48;
  }
}

@media (max-width: 767px) {
  .navbar-expand-md .navbar-nav .show .dropdown-menu .dropdown-item {
    color: #232424;
  }
  .navbar-expand-md .navbar-nav .show .dropdown-menu .dropdown-item:hover,
  .navbar-expand-md .navbar-nav .show .dropdown-menu .dropdown-item:focus {
    color: #ffffff;
  }
  .navbar-expand-md .navbar-nav .show .dropdown-menu .dropdown-item.active {
    color: #ffffff;
    background-color: #de6b48;
  }
}

@media (max-width: 991px) {
  .navbar-expand-lg .navbar-nav .show .dropdown-menu .dropdown-item {
    color: #232424;
  }
  .navbar-expand-lg .navbar-nav .show .dropdown-menu .dropdown-item:hover,
  .navbar-expand-lg .navbar-nav .show .dropdown-menu .dropdown-item:focus {
    color: #ffffff;
  }
  .navbar-expand-lg .navbar-nav .show .dropdown-menu .dropdown-item.active {
    color: #ffffff;
    background-color: #de6b48;
  }
}

@media (max-width: 1199px) {
  .navbar-expand-xl .navbar-nav .show .dropdown-menu .dropdown-item {
    color: #232424;
  }
  .navbar-expand-xl .navbar-nav .show .dropdown-menu .dropdown-item:hover,
  .navbar-expand-xl .navbar-nav .show .dropdown-menu .dropdown-item:focus {
    color: #ffffff;
  }
  .navbar-expand-xl .navbar-nav .show .dropdown-menu .dropdown-item.active {
    color: #ffffff;
    background-color: #de6b48;
  }
}

.navbar-expand .navbar-nav .show .dropdown-menu .dropdown-item {
  color: #232424;
}
.navbar-expand .navbar-nav .show .dropdown-menu .dropdown-item:hover,
.navbar-expand .navbar-nav .show .dropdown-menu .dropdown-item:focus {
  color: #ffffff;
}
.navbar-expand .navbar-nav .show .dropdown-menu .dropdown-item.active {
  color: #ffffff;
  background-color: #de6b48;
} */