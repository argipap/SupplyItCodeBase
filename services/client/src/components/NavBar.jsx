import React from 'react';
import {Link} from 'react-router-dom';

const NavBar = (props) => (
    // eslint-disable-next-line
    <nav className="navbar is-dark" role="navigation" aria-label="main navigation">
        <section className="container">
            <div className="navbar-brand">
                <strong className="navbar-item">{props.title}</strong>
                <span
                    className="nav-toggle navbar-burger"
                    onClick={() => {
                        let toggle = document.querySelector(".nav-toggle");
                        let menu = document.querySelector(".navbar-menu");
                        toggle.classList.toggle("is-active");
                        menu.classList.toggle("is-active");
                    }}>
          <span></span>
          <span></span>
          <span></span>
        </span>
            </div>
            <div className="navbar-menu">
                <div className="navbar-start">
                    <Link to="/" className="navbar-item">HOME</Link>
                    {!props.isAuthenticated &&
                    <Link to="/getStarted" className="navbar-item">GET STARTED</Link>
                    }
                    {props.isAuthenticated &&
                    <Link to="/status" className="navbar-item">USER STATUS</Link>
                    }
                    <Link to="/howItWorks" className="navbar-item">HOW IT WORKS</Link>
                    {!props.isAuthenticated &&
                    <Link to="/becomeSupplier" className="navbar-item">BECOME A SUPPLIER</Link>
                    }
                </div>
                <div className="navbar-end">
                    {!props.isAuthenticated &&
                    <div className="navbar-item">
                        <Link to="/login" className="button is-link">Sign In</Link>
                    </div>
                    }
                    {props.isAuthenticated &&
                    <Link to="/logout" className="navbar-item">Log Out</Link>
                    }
                </div>
            </div>
        </section>
    </nav>
);

export default NavBar;