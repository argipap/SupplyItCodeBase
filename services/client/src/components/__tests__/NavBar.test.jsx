import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {MemoryRouter as Router} from 'react-router-dom';
import routeData from 'react-router';


import NavBar from '../NavBar';

const title = 'SupplyIT logo';
const mockLocation = {
    pathname: '/login',
    hash: '',
    search: '',
    state: '',
};

const isAuthenticated = jest.fn();
const logoutUser = jest.fn();


beforeEach(() => {
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
});

test('NavBar renders properly', () => {
    const component = <Router><NavBar title={title} isAuthenticated={isAuthenticated} logoutUser={logoutUser}/></Router>;

    const wrapper = mount(component);
    const element = wrapper.find('nav[role="navigation"]');
    expect(element.length).toBe(1);
    const img = wrapper.find('img');
    expect(img.get(0).props['alt']).toBe(title);
});

test('NavBar renders a snapshot properly', () => {
    const tree = renderer.create(
        <Router location="/"><NavBar title={title} isAuthenticated={isAuthenticated} logoutUser={logoutUser}/></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});