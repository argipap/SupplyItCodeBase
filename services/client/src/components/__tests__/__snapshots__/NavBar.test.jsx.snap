// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders a snapshot properly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar navbar-expand-lg navbar-light bg-light sticky-top"
  navbar="light"
  role="navigation"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <img
      alt="SupplyIT logo"
      className="d-inline-block align-top"
      src="logo.png"
      width="110"
    />
  </a>
  <button
    aria-controls="navbar-bar"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="navbar-bar"
  >
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="active nav-link"
        data-rb-event-key="/"
        href="/"
        onClick={[Function]}
        onKeyDown={[Function]}
        style={Object {}}
      >
        Αρχική
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/#getstarted"
        href="/#getstarted"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Δοκιμάστε
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/#howitworks"
        href="/#howitworks"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Πως δουλεύει
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/becomeSupplier"
        href="/becomeSupplier"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Γίνετε προμηθευτής
      </a>
      <a
        className="btn-square btn btn-outline-success"
        data-testid="nav-login"
        href="/login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Σύνδεση
      </a>
    </div>
  </div>
</nav>
`;
